Common/src/gRPC/Services/Submitter.cs

```diff
@@ -133,27 +133,50 @@ public class Submitter : ISubmitter
                                  cancellationToken)
                     .ConfigureAwait(false);
   }
-
   /// <inheritdoc />
   public async Task<(IEnumerable<Storage.TaskRequest> requests, int priority)> CreateTasks(string                        sessionId,
                                                                                            string                        parentTaskId,
                                                                                            TaskOptions                   options,
                                                                                            IAsyncEnumerable<TaskRequest> taskRequests,
                                                                                            CancellationToken             cancellationToken)
+  {
+    var sessionData = await sessionTable_.GetSession(sessionId,
+                                                     cancellationToken)
+                                         .ConfigureAwait(false);
+    return await this.CreateTasks(sessionData,
+                                  parentTaskId,
+                                  options,
+                                  taskRequests,
+                                  cancellationToken)
+                     .ConfigureAwait(false);
+  }
+
+  /// <inheritdoc />
+  public async Task<(IEnumerable<Storage.TaskRequest> requests, int priority)> CreateTasks(SessionData                   sessionData,
+                                                                                           string                        parentTaskId,
+                                                                                           TaskOptions                   options,
+                                                                                           IAsyncEnumerable<TaskRequest> taskRequests,
+                                                                                           CancellationToken             cancellationToken)
   {
     using var logFunction = logger_.LogFunction(parentTaskId);
     using var activity    = activitySource_.StartActivity($"{nameof(CreateTasks)}");
-    using var sessionScope = logger_.BeginPropertyScope(("Session", sessionId),
+    using var sessionScope = logger_.BeginPropertyScope(("Session", sessionData.SessionId),
```

Comment:
The partition can be deduce from the Task?

Reply:
added PartitionId to scope